pkg/reconciler/v1alpha1/pipelinerun/resources/conditionresolution.go

```diff
@@ -0,0 +1,111 @@
+/*
+ *
+ * Copyright 2019 The Tekton Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package resources
+
+import (
+	"github.com/knative/pkg/apis"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+)
+
+const (
+	unnamedCheckNamePrefix = "condition-check-"
+)
+
+// GetCondition is a function used to retrieve PipelineConditions.
+type GetCondition func(string) (*v1alpha1.Condition, error)
+
+type ResolvedConditionCheck struct {
+	ConditionCheckName string
+	Condition          *v1alpha1.Condition
+	ConditionCheck     *v1alpha1.ConditionCheck
+}
+
+type TaskConditionCheckState []*ResolvedConditionCheck
+
+func (state TaskConditionCheckState) HasStarted() bool {
+	hasStarted := true
+	for _, j := range state {
+		if j.ConditionCheck == nil {
+			hasStarted = false
+		}
+	}
+	return hasStarted
+}
+
+func (state TaskConditionCheckState) IsComplete() bool {
+	if !state.HasStarted() {
+		return false
+	}
+	isDone := true
+	for _, rcc := range state {
+		isDone = isDone && !rcc.ConditionCheck.Status.GetCondition(apis.ConditionSucceeded).IsUnknown()
+	}
+	return isDone
+}
+
+func (state TaskConditionCheckState) IsSuccess() bool {
```

Comment:
could a docstring be added for each variable + function exported in this file?

Reply:
done